x-pack/test/observability_ai_assistant_functional/tests/feature_controls/assistant_security.spec.ts

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../ftr_provider_context';
+import {
+  createLlmProxy,
+  LlmProxy,
+} from '../../../observability_ai_assistant_api_integration/common/create_llm_proxy';
+import { createConnector, deleteConnectors } from '../../common/connectors';
+
+export default function ({ getPageObjects, getService }: FtrProviderContext) {
+  const log = getService('log');
+  const supertest = getService('supertest');
+  const security = getService('security');
+  const PageObjects = getPageObjects(['common', 'error', 'navigationalSearch', 'security']);
+  const ui = getService('observabilityAIAssistantUI');
+  const testSubjects = getService('testSubjects');
+
+  describe('ai assistant security', () => {
+    describe('ai assistant all privileges', () => {
+      before(async () => {
+        await security.role.create('ai_assistant_role', {
+          kibana: [
+            {
+              feature: {
+                // need some obs app or obs menu wont show where we can click on AI Assistant
+                infrastructure: ['all'],
+                observabilityAIAssistant: ['all'],
+                // requires connectors to chat
+                actions: ['read'],
+              },
+              spaces: ['*'],
+            },
+          ],
+        });
+
+        await security.user.create('ai_assistant_user', {
+          password: 'ai_assistant_user-password',
+          roles: ['ai_assistant_role'],
+          full_name: 'test user',
+        });
+
+        await PageObjects.security.forceLogout();
+
+        await PageObjects.security.login('ai_assistant_user', 'ai_assistant_user-password', {
+          expectSpaceSelector: false,
+        });
+      });
+
+      after(async () => {
+        // NOTE: Logout needs to happen before anything else to avoid flaky behavior
+        await PageObjects.security.forceLogout();
+        await Promise.all([
+          security.role.delete('ai_assistant_role'),
+          security.user.delete('ai_assistant_user'),
```

Comment:
Perhaps make these constants defined together with the helper mentioned above

```suggestion
          security.role.delete(AI_ASSISTANT_ROLE_NAME),
          security.user.delete(AI_ASSISTANT_USER_NAME),
```